unified-installer/astra-unified-installer.sh

```diff
@@ -188,6 +212,90 @@ get_configs() {
     CONNECTOR_AUTOSUPPORT_URL="${CONNECTOR_AUTOSUPPORT_URL:-$__PRODUCTION_AUTOSUPPORT_URL}"
 }
 
+print_help() {
+    cat <<EOF
+Here is a list of available options and environment variables for configuring the Astra cluster installation script.
+Values which are required and have no default value are marked with '*'.
+
+Options:
+  --help/-h    Show this message and exit.
+  --required   Prints all required environment variables which do not have a default value.
```

Comment:
another way I've seen this done is to simply mark them as REQUIRED in the list we print out with --help. Have you considered that option?

Reply:
Currently they're marked with `*`, but it's essentially the same thing with a different marker. My concern does remain that there's so many environment variables, it's hard to really tell at a glance what a user _needs_ to provide, even with a more obvious marker (such as REQUIRED).

Here's the changes I had in mind:
```
Usage: 
  ENV_VAR_1="value" ENV_VAR_2="value" ./astra-unified-installer.sh [options]

Options:
  --help/-h    Show this message and exit.

Required Environment Variables:
  KUBECONFIG                        Path to the KUBECONFIG for the cluster you'd like to manage. Required.
  ASTRA_API_TOKEN                   The Astra API token. Required, provided by the Astra Control UI.
  ASTRA_CONTROL_URL                 The Astra Control URL. Required, provided by the Astra Control UI.
  ASTRA_ACCOUNT_ID                  The Astra account ID. Required, provided by the Astra Control UI.

Optional Environment Variables:
  ----- Script Functionality
  CONFIG_FILE                        Path to a configuration file. Overrides environment variables specified via command line.
  DRY_RUN                            Skips applying generated resources when set to true. Default is false.
  SKIP_IMAGE_CHECK                   Skips checking if images exist when set to true. Default is false.
  SKIP_ASTRA_CHECK                   Skips all checks requiring a connection to Astra Control when set to true. Default is false.
  SKIP_TLS_VALIDATION                Skips TLS validation for all requests to Astra Control, including the Connector (unless CONNECTOR_SKIP_TLS_VALIDATION is set) when set to true. Default is false.
  DISABLE_PROMPTS                    Skips all prompts, answering 'yes' by default when set to true. Default is false.
  DO_NOT_MODIFY_EXISTING_TRIDENT     Prevents any and all modification to the existing Trident installation (if any). Required if DISABLE_PROMPTS is true, otherwise defaults to false.

  ----- General Configuration
  COMPONENTS                         Determines what will be installed/upgraded. Default is ALL_ASTRA_CONTROL.
  IMAGE_PULL_SECRET                  Image pull secret for the Docker registry.
  NAMESPACE                          Overrides EVERY resource's namespace (for fresh installs only, not upgrades).
  LABELS                             Labels to be added to the generated resources (disclaimer: does not apply labels to resources created by the operators).
  RESOURCE_LIMITS_PRESET             Resource limit preset to use. Overridden by RESOURCE_LIMITS_CUSTOM_CPU and RESOURCE_LIMITS_CUSTOM_MEMORY (disclaimer: resource limits not currently supported by Trident).
  RESOURCE_LIMITS_CUSTOM_CPU         Custom CPU resource limit. Plain number.
  RESOURCE_LIMITS_CUSTOM_MEMORY      Custom Memory resource limit (in 'Gi')'. Plain number (i.e. do not include the 'Gi').

  ----- Image Configuration
    Image configuration is separated into three distinct parts:
      - Registry ("private.registry.io")
      - Repository ("some/repository/path/trident")
      - Tag ("24.02")

    Given the following images :
      - Trident Operator:    private.registry.io/some/repository/path/trident-operator:24.02
      - Trident Autosupport: private.registry.io/some/repository/path/trident-autosupport:24.02
      - Trident:             private.registry.io/some/repository/path/trident:24.02
      - Trident ACP:         private.registry.io/some/repository/path/trident-acp:24.02
      - Connector Operator:  private.registry.io/some/repository/path/astra-connector-operator:202405211614-main

    An appropriate configuration might be:
      - IMAGE_REGISTRY="private.registry.io"
      - IMAGE_BASE_REPO="some/repository/path"
      - TRIDENT_IMAGE_TAG="24.02"
      - CONNECTOR_OPERATOR_IMAGE_TAG="202405211614-main"

  IMAGE_REGISTRY                     The default Docker registry to pull all images from. Overridden by values below.
  TRIDENT_OPERATOR_IMAGE_REGISTRY    The Docker registry to pull the Trident Operator image from.
  TRIDENT_AUTOSUPPORT_IMAGE_REGISTRY The Docker registry to pull the Trident Autosupport image from.
  TRIDENT_IMAGE_REGISTRY             The Docker registry to pull the Trident image from.
  CONNECTOR_OPERATOR_IMAGE_REGISTRY  The Docker registry to pull the Connector Operator image from.
  CONNECTOR_IMAGE_REGISTRY           The Docker registry to pull the Connector image from.
  NEPTUNE_IMAGE_REGISTRY             The Docker registry to pull the Neptune image from.
  TRIDENT_ACP_IMAGE_REGISTRY         The Docker registry to pull the Trident ACP image from.

  IMAGE_BASE_REPO                    The default base repository path (i.e. excluding the image name) for all images. Overridden by values below.
  TRIDENT_OPERATOR_IMAGE_REPO        The repository path for the Trident Operator image.
  TRIDENT_AUTOSUPPORT_IMAGE_REPO     The repository path for the Trident Autosupport image.
  TRIDENT_IMAGE_REPO                 The repository path for the Trident image.
  CONNECTOR_OPERATOR_IMAGE_REPO      The repository path for the Connector Operator image.
  CONNECTOR_IMAGE_REPO               The repository path for the Connector image.
  NEPTUNE_IMAGE_REPO                 The repository path for the Neptune image.
  TRIDENT_ACP_IMAGE_REPO             The repository path for the Trident ACP image.

  TRIDENT_IMAGE_TAG                  The tag for the Trident image. Overridden by other 'TRIDENT_' values below.
  TRIDENT_OPERATOR_IMAGE_TAG         The tag for the Trident Operator image.
  TRIDENT_AUTOSUPPORT_IMAGE_TAG      The tag for the Trident Autosupport image.
  TRIDENT_ACP_IMAGE_TAG              The tag for the Trident ACP image.
  CONNECTOR_OPERATOR_IMAGE_TAG       The tag for the Connector Operator image.

  ----- Connector Configuration
  ASTRA_CLOUD_ID                     The Astra cloud ID. If not provided,
  ASTRA_CLUSTER_ID                   The Astra cluster ID. Required, provided by the Astra Control UI.
  CONNECTOR_HOST_ALIAS_IP            Sets a host alias for the Astra Connector pod.
  CONNECTOR_SKIP_TLS_VALIDATION      (WARNING: Not for production use!) Skips TLS validation for the Connector's requests to Astra Control if set to true.
  CONNECTOR_AUTOSUPPORT_ENROLLED     Enrolls the Connector in autosupport if set to true. Default is false.

```

Reply:
ah, my bad. missed the first line printed above 🤦 

I like the new organization your proposed. Either works for me.